ansible_base/authentication/authenticator_plugins/oidc.py

```diff
@@ -215,3 +217,37 @@ def extra_data(self, user, backend, response, *args, **kwargs):
 
     def get_user_groups(self, extra_groups=[]):
         return extra_groups
+
+    def public_key(self):
+        key = self.setting("PUBLIC_KEY")
+        return (
+            "\n".join(
+                [
+                    "-----BEGIN PUBLIC KEY-----",
+                    key,
+                    "-----END PUBLIC KEY-----",
+                ]
+            )
+            if key
+            else None
+        )
+
+    def user_data(self, access_token, *args, **kwargs):
+        user_data = self.request(self.userinfo_url(), headers={"Authorization": f"Bearer {access_token}"})
+        if user_data.headers["Content-Type"] == "application/jwt":
+            pk = self.public_key()
+            if not pk:
+                logger.error(_("OIDC client sent encrypted user info response, but no public key found."))
```

Comment:
As far as I have seen, it is more common that it is not encrypted and/or signed.  Even in RH SSO I believe it comes back unencrypted.  So yeah, it can and usually does come back unencrypted for sure.

The standard is here and 5.3.2 just says that the server may encrypt/sign it.

https://openid.net/specs/openid-connect-core-1_0.html#UserInfo

But yeah, it is strange that social auth doesn't support that, especially considering that their own keycloak implementation does.  Keycloak itself though by default does not encrypt/sign that response and you have to go into the advanced settings to enable it..

James Tanner added the PUBLIC_KEY to the dab oidc config but it wasn't doing anything yet ("The public key from your IDP. Only necessary if using keycloak for OIDC.").  It doesn't come from social auth and would be needed if they were to do this jwt unpacking.

Reply:
So what happens in this code if the JWT is decrypted? Or would this code just not get called if its not encrypted?

Reply:
Yeah, they have application/json content type rather than jwt type, so they just get returned by the last line.